### PR TITLE
Add support for X-Remix-Reload-Document header

### PR DESCRIPTION
**Pull request by robbtraister**
This PR is paired with https://github.com/remix-run/react-router/pull/10705 to add support for passing along the `X-Remix-Reload-Document` header.

[Open Development discussion](https://github.com/remix-run/react-router/discussions/9859)

### CODE REVIEW

**Comment by zackeryg** [comment]
Any idea when this is going to land? Need any help getting this in

**Comment by gerbyzation** [comment]
@zackeryg see https://github.com/remix-run/react-router/pull/10705#issuecomment-1662868188

**Comment by brophdawg11** [comment]
Thanks @robbtraister!  We updated to an experimental react router version in https://github.com/remix-run/remix/pull/7040 so this is good to merge to dev and ship in Remix v2.  I added an integration test as well so I'll get this merge once CI passes.

File: packages/remix-react/routes.tsx [hunk 1]
```diff
@@ -258,5 +258,9 @@ function getRedirect(response: Response): Response {
   if (revalidate) {
     headers["X-Remix-Revalidate"] = revalidate;
   }
+  let reloadDocument = response.headers.get("X-Remix-Reload-Document");
+  if (reloadDocument) {
+    headers["X-Remix-Reload-Document"] = reloadDocument;
+  }
```
Comment (robbtraister) [inline packages/remix-react/routes.tsx:264]: @brophdawg11 do you think it's worth refactoring this into an array + for-loop (or .forEach or .reduce)?

something like:
```ts
let HEADERS_TO_PRESERVE = ["X-Remix-Revalidate", "X-Remix-Reload-Document"];

...

  for (let headerName of HEADERS_TO_PRESERVE) {
    let headerValue = response.headers.get(headerName);
    if (headerValue) {
      headers[headerName] = headerValue;
    }
  }
```

Comment (brophdawg11) [reply packages/remix-react/routes.tsx:264]: nah not yet for the 2nd header - something something kent c dodds avoid hasty abstractions :)

**Pull request merged**